src/main/Main.hs

```diff
@@ -773,7 +773,7 @@ execCmd ExecOpts {..} go@GlobalOpts{..} =
                 -- Add RTS options to arguments
                 let argsWithRts args = if null eoRtsOptions 
                             then args :: [String]
-                            else args ++ ["+RTS"] ++ eoRtsOptions ++ ["-RTS"]
+                            else args ++ ["+RTS"] ++ concatMap words eoRtsOptions ++ ["-RTS"]
```

Comment:
After staring at this for a while, I think it's OK. That usage gets passed to Cabal inside a `--ghc-options` flag itself, so therefore it becomes Cabal's responsibility to parse it into individual flags to hand off to GHC. Though to be fair I'm not 100% certain. I've asked some others to take a look.